src/tezos_fmt_plug.c

```diff
@@ -0,0 +1,343 @@
+	char *ctcopy = strdup(ciphertext);
+	char *keeptr = ctcopy;
+	char *p;
+	int i;
+
+	memset(&cs, 0, SALT_SIZE);
+	ctcopy += TAG_LENGTH;
+	p = strtokm(ctcopy, "*");
+	cs.type = atoi(p);
+	p = strtokm(NULL, "*");
+	cs.iterations = atoi(p);
+	p = strtokm(NULL, "*");
+	strcpy(cs.mnemonic, p);
+	cs.mnemonic_length = strlen(p);
+	p = strtokm(NULL, "*");
+	strcpy(cs.email, p);
+	cs.email_length = strlen(p);
+	p = strtokm(NULL, "*");
+	strcpy(cs.address, p);
+	p = strtokm(NULL, "*");
+	cs.raw_address_length = strlen(p) / 2;
+	for (i = 0; i < cs.raw_address_length; i++)
+		cs.raw_address[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
+
+	MEM_FREE(keeptr);
+	return &cs;
+}
+
+static void set_salt(void *salt)
+{
+	cur_salt = (struct custom_salt *)salt;
+}
+
+static void tezos_set_key(char *key, int index)
+{
+	strnzcpy(saved_key[index], key, PLAINTEXT_LENGTH + 1);
+}
+
+static char *get_key(int index)
+{
+	return saved_key[index];
+}
+
+static int crypt_all(int *pcount, struct db_salt *salt)
+{
+	const int count = *pcount;
+	int index;
+
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+	for (index = 0; index < count; index += MIN_KEYS_PER_CRYPT) {
+		unsigned char seed[MIN_KEYS_PER_CRYPT][64];
+		char salt[MIN_KEYS_PER_CRYPT][16 + 256 + PLAINTEXT_LENGTH];
+		int i;
+#ifdef SIMD_COEF_64
+		int lens[MIN_KEYS_PER_CRYPT];
+		int slens[MIN_KEYS_PER_CRYPT];
+		unsigned char *pin[MIN_KEYS_PER_CRYPT], *pout[MIN_KEYS_PER_CRYPT];
+		unsigned char *sin[MIN_KEYS_PER_CRYPT];
+#endif
+		// create varying salt(s)
+		for (i = 0; i < MIN_KEYS_PER_CRYPT; ++i) {
+			memcpy(salt[i], "mnemonic", 8);
+			memcpy(salt[i] + 8, cur_salt->email, cur_salt->email_length + 1);
+			strcat(salt[i], saved_key[index+i]);
+		}
+
+		// kdf
+#ifdef SIMD_COEF_64
+		for (i = 0; i < MIN_KEYS_PER_CRYPT; ++i) {
+			lens[i] = cur_salt->mnemonic_length;
+			pin[i] = (unsigned char*)cur_salt->mnemonic;
+			sin[i] = (unsigned char*)salt[i];
+			pout[i] = seed[i];
+			slens[i] = strlen(salt[i]);
+		}
+		pbkdf2_sha512_sse_varying_salt((const unsigned char**)pin, lens, sin, slens, 2048, pout, 64, 0);
+#else
+		for (i = 0; i < MIN_KEYS_PER_CRYPT; ++i)
+			pbkdf2_sha512((unsigned char*)cur_salt->mnemonic,
+					cur_salt->mnemonic_length, (unsigned char*)salt[i], strlen(salt[i]), 2048,
+					seed[i], 64, 0);
+#endif
+		for (i = 0; i < MIN_KEYS_PER_CRYPT; ++i) {
+			unsigned char buffer[20];
+			ed25519_public_key pk;
+			ed25519_secret_key sk;
+
+			// asymmetric stuff
+			memcpy(sk, seed[i], 32);
+			ed25519_publickey(sk, pk);
+
+			blake2b((uint8_t *)buffer, (unsigned char*)pk, NULL, 20, 32, 0); // pk is pkh (pubkey hash)
+
+			if (memmem(cur_salt->raw_address, cur_salt->raw_address_length, (void*)buffer, 8)) {
+				cracked[index+i] = 1;
+			} else {
+				cracked[index+i] = 0;
+			}
+		}
+	}
+
+	return count;
+}
+
+static int cmp_all(void *binary, int count)
+{
+	int index;
+
+	for (index = 0; index < count; index++)
+		if (cracked[index])
+			return 1;
```

Comment:
Maybe introduce `any_cracked` for some speedup, like some other non-hash formats have.